Themes/default/languages/Help.english.php

```diff
@@ -381,7 +383,12 @@
 $helptxt['preview_characters'] = 'This setting sets the number of available characters for the first and last message topic preview.';
 $helptxt['quote_expand'] = 'This setting allows large quotes to be cutoff and expanded to their full length by the user. The setting indicates the minimum height a quote needs to have for it to be collapsed. Quotes smaller than this setting won\'t be affected';
 $helptxt['posts_require_captcha'] = 'This setting will force users to pass anti-spam bot verification each time they make a post to a board. Only users with a post count below the number set will need to enter the code - this should help combat automated spamming scripts.';
-$helptxt['enableSpellChecking'] = 'Enable spell checking. You MUST have the pspell or enchant library installed on your server and your PHP configuration set up to use the installed library. Your server ' . ((function_exists('pspell_new') || function_exists('enchant_broker_init')) ? 'DOES' : 'DOES NOT') . ' appear to have this set up.';
+// argument(s): either $helptxt['enableSpellCheckingSupported'] or $helptxt['enableSpellCheckingUnsupported'], depending on whether the server supports one of the libs
+$helptxt['enableSpellChecking'] = 'Enable spell checking. You MUST have the pspell or enchant library installed on your server and your PHP configuration set up to use the installed library. Your server %1$s appear to have this set up.';
+// is conditionally used as part of $helptxt['enableSpellChecking']
+$helptxt['enableSpellCheckingSupported'] = 'DOES';
+// is conditionally used as part of $helptxt['enableSpellChecking']
+$helptxt['enableSpellCheckingUnsupported'] = 'DOES NOT';
```

Comment:
This bothers me a little xD

We really shouldn't be breaking phrases up like that, they won't be able to be properly translated this way.

even the way it was before the edit was wrong, what we should have is two text strings

One for when it is supported
```php
$helptxt['enableSpellChecking'] = 'Enable spell checking. You MUST have the pspell or enchant library installed on your server and your PHP configuration set up to use the installed library. Your server DOES appear to have this set up.';
```

The other for when it is not supported
```
$helptxt['enableSpellChecking'] = 'Enable spell checking. You MUST have the pspell or enchant library installed on your server and your PHP configuration set up to use the installed library. Your server DOES NOT appear to have this set up.';
```

It should not matter whether a big part of the string is practically duplicated, this is a better approach for the sake of sane translation, other languages translate very differently and thus imho we should not control the string with words this way.

Reply:
Or even better we should just separate the `Your server DOES/NOT appear to have this set up.` into two strings so the first bit won't be duplicated this way. and then we simply concatenate them in the source file.

Reply:
I attempted to move in that direction in f0afe70ad7cad3b7ce6638715f92f4aeaa9d4868, but I'm unsure how to change the [code](https://github.com/SimpleMachines/SMF2.1/pull/6121/commits/b75bac4b4c87bb30c2c2c5c1e1973e5281263bc9), since a single `$helptxt` variable has to be passed. Do you want it to be changed, so that instead of `enableSpellChecking`, `enableSpellChecking{Supported,Unsupported}` is being passed?

Reply:
The code actually works just fine with the change, `sprintf()` either injects `$helptxt['enableSpellCheckingSupported']` or `$helptxt['enableSpellCheckingUnsupported']` at the end of the text string which is exactly what we want.

unless I didn't understand what you meant ?

Reply:
What I meant was that `Sources/Help.php` is currently passing  "`enableSpellChecking`" to `Themes/default/Help.template.php`, and because I'm not 100% sure how to change that in `Help.php`, I still left it in the template until now. Will try to fix that today.